platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -115,53 +170,62 @@ public CFMetaData getSSTableMetadata(Descriptor desc) throws Exception{
     /** Converting Cassandra's data type to Hillview data type */
     private static ColumnDescription getDescription(ColumnDefinition colDef) {
         String name = colDef.toString();
-        String colType = colDef.type.asCQL3Type().toString();
         ContentsKind kind;
-        switch (colType) {
-            case "boolean":
-            case "ascii":
-            case "inet":
-            case "text":
-            case "timeuuid":
-            case "uuid":
+        CQL3Type type = colDef.type.asCQL3Type();
+        switch ((CQL3Type.Native) colDef.type.asCQL3Type()) {
+            case BOOLEAN:
+            case ASCII:
+            case INET:
+            case TEXT:
+            case TIMEUUID:
+            case UUID:
+            case DURATION:
+            case TIME:
+            case DATE:
```

Comment:
I've tried that, but the format of `Date, Time, and Duration` at Cassandra can't be converted directly to Java style. We must get the formating style before converting it. Especially, the formatting of Cassandra's `Duration` is unlike Java's duration, I have to convert, split, and parse the data to match Java's Duration.  It will be faster if client applies `filter` to these fields as needed. What do you think?